mini-modeline.el

```diff
@@ -177,14 +178,27 @@ When ARG is:
                   ;; write to minibuffer
                   (erase-buffer)
                   (when mini-modeline--cache
-                    (if (> (cdr mini-modeline--cache) 1)
-                        (window-resize (minibuffer-window mini-modeline-frame)
-                                       (- (cdr mini-modeline--cache)
-                                          (window-height (minibuffer-window mini-modeline-frame)))))
-                    (insert (car mini-modeline--cache))))))))
+                    (if (>= (cdr mini-modeline--cache) 1)
+                        (let ((resize-mini-windows t) ; let mini-modeline take control of mini-buffer size
+                              (height-increase (- (cdr mini-modeline--cache)
+                                                  (window-height (minibuffer-window mini-modeline-frame)))))
+                          (when (or (> height-increase 0)
+                                    ;; this is to prevent window flashing for consecutive multi-line message
+                                    (>= (float-time (time-since mini-modeline--last-change-size))
+                                        mini-modeline-echo-duration))
+                            (window-resize (minibuffer-window mini-modeline-frame)
+                                           height-increase)
+                            (setq mini-modeline--last-change-size (current-time)))))
+                      (insert (car mini-modeline--cache))))))))
```

Comment:
This seems not properly indented.

Reply:
I use Lispy for the indentation, I really don't have an idea on what it should be like.